Aikido.Zen.Core/Models/AgentContext.cs

```diff
@@ -123,9 +122,29 @@ public void Clear()
             _blockedUsers.Clear();
         }
 
-        public bool IsBlocked(User user, string ip, string endpoint, string userAgent)
+        public bool IsBlocked(User user, string ip, string endpoint, string userAgent, out string reason)
         {
-            return (user != null && IsUserBlocked(user.Id)) || _blockList.IsBlocked(ip, endpoint) || IsUserAgentBlocked(userAgent);
+            reason = null;
+            // if the ip is bypassed, we don't block the request
+            if (BlockList.IsIPBypassed(ip))
+            {
+                return true;
+            }
+            if (user != null && IsUserBlocked(user.Id))
+            {
+                reason = "User is blocked";
+                return true;
+            }
+            if (_blockList.IsBlocked(ip, endpoint, out reason))
+            {
+                return true;
+            }
+            if (IsUserAgentBlocked(userAgent))
+            {
+                reason = "User agent is blocked";
```

Comment:
We would want to show the same message as node for this one, https://github.com/AikidoSec/firewall-node/blob/10ed01d1f720bedceef70026fcf802dbcf04e898/library/sources/http-server/checkIfRequestIsBlocked.ts#L95

Reply:
Let's try to be as close as possible to the node responses here, because we could be giving away too much information to end-user.